### ISSUE TITLE
React Native support

### ISSUE THREAD

**Issue by zaguiini** [chakra-ui/chakra-ui#183]
First of all I'm sorry if this is not the correct place to do this.

So I'm seeing this UI library and it's astonishing. I didn't see such beauty in a while and I'm wondering if we could extend this exact lib to React Native.

I know it's a huge step and I don't expect it to be fast, but in my point of view it would be amazing since I don't know many libraries that has this kind of support.

What do you think? We can create a RFC or an umbrella issue/PR adding that possibility. I think it would be awesome to replicate the exact same styles to the mobile platform, since they are amazing.

Thanks in advance!

**Comment by segunadebayo** [comment]
This sounds interesting @zaguiini. It's not in the pipeline at the moment but it's something worth considering. Feel free to start some work on it if you're interested, maybe pick a few components to implement in React Native.

I don't know React Native at the moment but I'd be happy to look through the code once you have a PR.

**Comment by zaguiini** [comment]
Could you pin this issue so more people interested on it can see it?

**Comment by willianhf** [comment]
Would be nice use this library in React Native!
@zaguiini started implementing something? I'd be like to help!

**Comment by zaguiini** [comment]
No, I didn't. I'm not having much time lately, sorry. But if you start something, I'll help as much as I can.

**Comment by zaguiini** [comment]
Messing around with it. Some progress at least!

![image](https://user-images.githubusercontent.com/26530524/69390012-9318b680-0cac-11ea-8633-f15ed0da391e.png)

[Here](https://github.com/chakra-ui/chakra-ui/compare/master...zaguiini:master)* is what I did. Excited??? 😄 

* Of course this is not how I'm going to implement the `p` property for example... I just made it work with validation purposes. Maybe we'll make something more elaborated there. 😉 

@willianhf we could chat. How can I reach you? Slack? Skype?

**Comment by coodoo** [comment]
Making chakra cross platform is indeed a very intriguing idea, has anyone considered to build it on top of react-native-web so that it supports both web and mobile devices out of the box?

https://github.com/necolas/react-native-web

**Comment by willianhf** [comment]
@zaguiini can it be by discord? I usually use it, that's my tag `jet#5723`

**Comment by zaguiini** [comment]
> Making chakra cross platform is indeed a very intriguing idea, has anyone considered to build it on top of react-native-web so that it supports both web and mobile devices out of the box?
> 
> https://github.com/necolas/react-native-web

The problem with it is that we would need to rewrite the whole codebase again? Or is it incremental?

**Comment by steida** [comment]
RNW has own css-in-js, so yes, whole lib would have to be rewrittten.

**Comment by segunadebayo** [comment]
In the TypeScript implementation, I'm extracting most of the component logic into hooks so it can be consumed in React Native Web and React (Web). Perhaps, it might help.

**Comment by steida** [comment]
@segunadebayo It definitely helps, but as I said, the point of RNW is its API, both components and CSS.

**Comment by zaguiini** [comment]
As we have a monorepo, the approach I was thinking was to have a package to the component logic, maybe rename `@chakra-ui/core` to `@chakra-ui/web` and create a package called `native` which will use the logic from the sibling package and will have the styles created using emotion and styled-system, built with TypeScript from ground up (we might use TSDX for that).

Thoughts @segunadebayo ? I think that renaming from `core` to `web` might look tough but IMO that's a better option than having `core` and `native` since it becomes ambiguous... But if you don't like it, we'll leave it as it is

cc @willianhf @steida

**Comment by willianhf** [comment]
Sounds nice to me, but rename package could be a hindrance.

**Comment by zaguiini** [comment]
Just FYI: I'm already doing something along the lines of Chakra for RN, working on it. But... I'm not using `emotion` as it doesn't have typings for TS. I'm using `styled-components`, but the result is similar:

![image](https://user-images.githubusercontent.com/26530524/69889678-0d5ec180-12d1-11ea-8a00-39f2bfc2e4fc.png)

**Comment by steida** [comment]
I really don't understand why people use styled-components abstraction over perfect and typed React Native Web. OK, I know why. They believe untyped strings soup is better than typed JSON because it's sooooo familiar.

**Comment by ProsperDevelopment** [comment]
This is something my team also would be interested in - we have mobile apps coming later in the pipe, at it would be a-some to be able to use Chakra UI there as well.

**Comment by rahsheen** [comment]
Any updates on this?

**Comment by rynatk** [comment]
I'd be willing to help with this, and plan to do so in my spare time anyways. I use styled-system on every RN project I'm on these days and would love to leverage chakra as a consistent out-of-the-box setup.

**Comment by segunadebayo** [comment]
@zaguiini. Do you have any updates?

**Comment by matepapp** [comment]
@rynatk is styled-system compatible with React Native 😮?

**Comment by zaguiini** [comment]
@segunadebayo sorry, I'm having little to no time to work on this one :/ Life has been busy lately...

The first step would be to create `.native.js` files to deal with Metro's resolution

**Comment by segunadebayo** [comment]
No worries @zaguiini. Take your time, I believe someone here might be able to pick up from where you stopped. It might be good to leave the link to your repo here to show your progress so far.

Thank you man

**Comment by ri7nz** [comment]
@matepapp Yes, styled-system it's work on react-native, but need some adjustment for `breakpoints`. I have project with `react-native` and `styled-system`, I start from September 2019.

**Comment by jsartisan** [comment]
Hi all, If anyone looking for a react native port of chakra-ui, do consider - https://magnus-ui.com/ :)
It is heavily inspired by chakra-ui 👍 
I am working on the docs heavily nowaday. Looking forward to feedbacks,


@ri7nz @segunadebayo  @zaguiini @rahsheen

**Comment by jsartisan** [comment]
Here is an example - https://snack.expo.io/@pawankumar2901/dribble-login-page

**Comment by c-moss-talk** [comment]
We have a cross-platform component library at Talkspace. One path I could see would be to:

1. Replace all tag usages of styled with [React Primitives ](https://github.com/lelandrichardson/react-primitives), which establish an interface between platforms. That change would look like this
```
  const Box = styled("div", {...
  
  // becomes

`const Box = styled(View)({...
```
and View is imported from React Native Web.

2. Create a proxy for the emotion library where

**Comment by segunadebayo** [comment]
Thanks guys, for your input on this.

We'll refer to this thread once the team is ready to add RN support. Closing this, for now, considering there's been no much progress.

**Comment by vonwao** [comment]
> Hi all, If anyone looking for a react native port of chakra-ui, do consider - https://magnus-ui.com/ :)
> It is heavily inspired by chakra-ui 👍
> I am working on the docs heavily nowadays. Looking forward to feedbacks,

@jsartisan How does magnus-ui compare to react-native-elements?  Maybe if you post a comparison it would help people make a better choice.  

I know it's totally unrelated to this issue and web, but I was thinking of using react-native-web with react-native-elements, so I was also wondering about that.   Seems like there is starting to be some crossover betwen web and native, which is exciting (learn once, use everywhere).

**Comment by jsartisan** [comment]
@vonwao That's a good question. The main difference comes in terms of Utilities ( Spacing, Typography, Border Radius, etc).  You can use easily customize the component using the utility props ( Check out this example -  https://magnus-ui.com/docs/button/#example-5 ).

**Comment by coodoo** [comment]
At first glance I really like `magnus-ui`'s approach, which looks a lot like `chakra-ui`, any plan to support `react-native-web` as well?

**Comment by jsartisan** [comment]
@coodoo I can add the support. Let me see how much work is required.

**Comment by nandorojo** [comment]
@jsartisan If `magnus-ui` uses packages supported by @expo / @expo web, React Native Web support should work out of the box.

**Comment by yareyaredesuyo** [comment]
Another library that supports Styled System.

https://github.com/Shopify/restyle

**Comment by nandorojo** [comment]
You can also check out `dripsy`, a package I'm working on. It's like `theme-ui` for React Native/React Native Web. It has responsive styles & theme values, and I’m finishing up support for server side rendering (like Next.js).

https://github.com/nandorojo/dripsy

**Comment by sorxrob** [comment]
Since Chakra's theme is inspired by Tailwind, maybe https://github.com/vadimdemedes/tailwind-rn can do for now

**Comment by sorxrob** [comment]
Since Chakra's theme is inspired by Tailwind, maybe https://github.com/vadimdemedes/tailwind-rn will do for now

**Comment by hems** [comment]
Would love to see chakra-ui for React Native!

**Comment by jsartisan** [comment]
@hems Checkout https://magnus-ui.com/

**Comment by redbar0n** [comment]
I also wanted Chakra UI for React Native. But, on further consideration, I'd rather want good touch / swipe / gesture support in Chakra UI in general.

**Comment by nandorojo** [comment]
If it helps, React Native's `GestureHandler` has a great API for this. It also is really good for web.

Example from @evanbacon: https://heads.netlify.app/

**Comment by redbar0n** [comment]
I guess the desire for React Native is made most relevant due to React Native Web. I think this [comment by the Dripsy founder in its readme](https://github.com/nandorojo/dripsy) was particularly relevant. Because it's so easy to forget!

> “Build once, deploy everywhere, is a great philosophy made possible by Expo Web/React Native Web. **A large impediment is responsive design.** React Native doesn't have media queries for styles, and trying to micmick it with JS turns into useState hell with a ton of conditionals (as you'll see below.) **While React Native has some great component libraries, it lacks a good design system that is responsive and themed**.”

To me it sounds like a further argument to rely on a web-first or hybrid approach, as mentioned, where only augmenting with native components or interactions where absolutely necessary, and on a view-by-view basis.

**Comment by coodoo** [comment]
Now that chakra v1 is out of the door, wondering has the core team gave another thought on supporting react native? This would truly be the last missing piece in the whole eco system and a great chance for chakra-ui to shine and dominate! 😉

cc @segunadebayo

**Comment by jsartisan** [comment]
You can check https://magnus-ui.com/ :D

It's similar to chakra UI for react-native.

Currently, I am working on responsive props that will work on web too :)

**Comment by coodoo** [comment]
Yup, having been keeping a close eye on magnus for a while now and very looking forward to the responsive props feature to kick in, that would make magnus the next best thing to `chakra-ui/native` 😅

**Comment by jsartisan** [comment]
@coodoo yea I am working on that. It should be completed pretty soon. Let's hope for the best.

**Comment by redbar0n** [comment]
To get mobile gesture support in Chakra UI, maybe it could be possible to use either of these?

https://github.com/aholachek/mobile-first-animation

https://hammerjs.github.io/

**Comment by mjcarnaje** [comment]
can you use charka ui in react nativee?

**Comment by MatteoAlberghini** [comment]
Hi! Are there any updates on this project? I have some mobile experience in both react native & native development for both iOS and Android, I would like to help :)

**Comment by nandorojo** [comment]
If you want React Native support, you could use my library [Moti](moti.fyi) to replace (many) Framer Motion use-cases.

**Comment by redbar0n** [comment]
@steida : I don't think people attempt to use an extra CSS-in-JS library like styled-components or emotion-js on top of RNW because they dislike the RNW solution, or prefer string interpolation instead of RNW's StyleSheet. I suspect they are just confused, like I was, and think they need a styling solution with RN/RNW, when in reality it is already baked in.

@mjcarnaje No, you can't use Chakra UI out of the box with React Native.

**Comment by redbar0n** [comment]
I wrote up a blog post, going into a bit more detail on the excellent points already made by @steida, since they might have been lost to some, as they were to me. 

I also compared Dripsy vs. Magnus UI. Which wasn't obvious because of their great similarities, and since I had to dive a bit into the workings of Fresnel to figure out the difference.

I also looked at some alternative solutions to getting Chakra UI on native mobile, via. either a WebView or Capacitor.

https://dev.to/redbar0n/how-to-use-chakraui-on-native-mobile-53hg

Hope it helps someone. :)

**Comment by binajmen** [comment]
![Screenshot 2021-05-01 at 11 17 40](https://user-images.githubusercontent.com/15611419/116777888-08c2ad80-aa6f-11eb-8390-bc8589f045d4.png)

🥳 🥳

**Comment by chanmathew** [comment]
> ![Screenshot 2021-05-01 at 11 17 40](https://user-images.githubusercontent.com/15611419/116777888-08c2ad80-aa6f-11eb-8390-bc8589f045d4.png)
> 
> 
> 
> 🥳 🥳 

Is this for real?

**Comment by binajmen** [comment]
You can double check on https://twitter.com/chakra_ui/status/1388033321213612032?s=21

However, "pretty soon" remains vague 🤔

**Comment by chanmathew** [comment]
Hey @segunadebayo @zaguiini, just wanted to ask if you guys are indeed working on React Native support as it appears from your Twitter? Our work is looking into a UI framework to adopt and Chakra is a great option, but would really need cross-platform support, really looking forward to this!

**Comment by zameschua** [comment]
+1, this is the only thing I want in the entire world, please I love you guys so much

**Comment by binajmen** [comment]
To be honest, I've been trying several UI libraries for my RN project, and none compared to the beauty, clarity and easiness of Chakra-UI. I really hope it arrives soon... 😭

**Comment by BaoPham92** [comment]
Dropping support. 

Got the license for premium, great product, and the team as well. Appreciate the videos you guys have pushing out, thats a great on ramp for chakra-ui.

Ready for chakra on RN 👌

**Comment by cpnvr** [comment]
Also would love to know if this is happening :)

**Comment by with-heart** [comment]
I can't speak to the React Native support timeline, but we're working on larger plans for restructuring Chakra to be able to support any framework, so there is definitely a lot happening in that direction :)

**Comment by redbar0n** [comment]
@with-heart If one could use Chakra UI with RN's `StyleSheet.create` and pass all styles inside such an object to Chakra components through a `style` prop, that would go a long way towards RN & RNW compatibility.

**Comment by redbar0n** [comment]
@with-heart FWIW, here's some more hopefully relevant info. For cross-platform support, Chakra would likely have to drop using `emotion`, for the same reason that [styled-components is problematic with React Native Web (RNW)](https://github.com/jsartisan/react-native-magnus/issues/59#issuecomment-855434897). Max Stoiber of styled-components also [said](https://twitter.com/mxstbr/status/873165476125343744?s=20):

> styled-components won't let you use the exact same styling code in React and ReactNative. react-primitives and/or react-native-web would be a better bet!

Unless using `StyleSheet.create` in [emotion with react-360](https://emotion.sh/docs/@emotion/native#usage-with-react-360). But [react-360](https://github.com/facebookarchive/react-360) is an abandoned project by Facebook..

react-primitives uses the same API as RNW, but extends it to support more targets than just Native and Web (importing React components into Figma etc.). It should be at the right level of abstraction, to enable cross-platform support:

> This library attempts to propose an ideal set of primitives around building React applications, regardless of Platform. In the future, this could be used as a shared interface among React and React Native components that don't use platform-specific APIs.
>...
> This library was largely inspired from the work done by Nicolas Gallager and his great work on the react-native-web library. 

https://github.com/lelandrichardson/react-primitives

**Comment by MohamedShatry** [comment]
Native bases's version 3 basically adopted Chakra's approach, and it's built for React Native. Check it out: https://docs.nativebase.io/

**Comment by HommeSauvage** [comment]
Native Base isn't that good, I'd even say it's a poor quality ChakraUI wannabe for RN. Check the kitchen sink and other docs/demos, you'll see that the quality of Chakra isn't there.

**Comment by wobsoriano** [comment]
> Native Base isn't that good, I'd even say it's a poor quality ChakraUI wannabe for RN. Check the kitchen sink and other docs/demos, you'll see that the quality of Chakra isn't there.

You're talking about the old version, am I right? The v3 docs looks nice tbf (comparing to older version)

**Comment by zameschua** [comment]
For people coming into this thread in the future, we're moving our stack to Shopify Restyle.
https://github.com/Shopify/restyle

Gives a lot of the benefits that I'm looking for (theming, typesafety etc) while being super flexible.

**Comment by nandorojo** [comment]
I maintain Dripsy, which targets web and native. Restyle looks really solid. I'm not sure if it's optimized for web. I don't think they use StyleSheet under the hood, which creates class names for you instead of inline styles. Dripsy does use StyleSheet for web.

However, Restyle has more type safety than Dripsy. It's a bit more cumbersome to setup, but it looks solid.

While NativeBase seems to have improved, I don't think their design is Chakra-level. The components aren't as consistent and good looking. But if you use NativeBase as unstyled components, and use your own styles, you could use the Chakra API which is nice.

I think the need for a Chakra for React Native (+Web) remains. 

Similarly, if anyone is working on a Radix UI for RN/RNW, please let me know.

**Comment by Mi3liX9** [comment]
Hello there, anything new?

**Comment by pierroo** [comment]
> Native base's version 3 basically adopted Chakra's approach, and it's built for React Native. Check it out: https://docs.nativebase.io/

Actually I recently discovered nativebase, too.
And I got really surprised at how "similar" it is to chakra ui; I didn't know who "got inspired" by whom until now.
Did anyone actually try nativebase for their react native app?

Even if their v3 might have improved, I believe just like @nandorojo that it is not yet close to chakra ui aesthetic.

But I am afraid that due to deadlines I would have to go with nativebase rather than chakra for RN just because I got used to chakra way of coding SO fast, I couldn't imagine going back the old way now.

So any kind of estimated delivery time would be very much appreciated. :')

Btw, if I got the chakra ui pro already, will this also include a potential chakra pro for RN when it is released?

**Comment by mirabledictu** [comment]
> > Native base's version 3 basically adopted Chakra's approach, and it's built for React Native. Check it out: https://docs.nativebase.io/
> 
> Actually I recently discovered nativebase, too.
> And I got really surprised at how "similar" it is to chakra ui; I didn't know who "got inspired" by whom until now.
> Did anyone actually try nativebase for their react native app?
> 
> Even if their v3 might have improved, I believe just like @nandorojo that it is not yet close to chakra ui aesthetic.
> 
> But I am afraid that due to deadlines I would have to go with nativebase rather than chakra for RN just because I got used to chakra way of coding SO fast, I couldn't imagine going back the old way now.
> 
> So any kind of estimated delivery time would be very much appreciated. :')
> 
> Btw, if I got the chakra ui pro already, will this also include a potential chakra pro for RN when it is released?

Send them a PR with complete features since you're in a rush

**Comment by itsmelion** [comment]
Well, I've been using styled-components in RN for a while, and seems to be all ok.
I imagine emotion should be aswell.
I believe the biggest challenge in chakra is regarding animations.

nonetheless i loved native-base idea, and will pick it for my next project. i think it will be easier to refactor in chakra if it ever gets released

**Comment by iamclaytonray** [comment]
i'm just grabbing the theme from chakra and putting in nativebase for now. if/when chakra releases a native lib, the transition should be relatively seamless. getting the aesthetic is relatively simple

**Comment by pierroo** [comment]
> i'm just grabbing the theme from chakra and putting in nativebase for now. if/when chakra releases a native lib, the transition should be relatively seamless. getting the aesthetic is relatively simple

That's interesting; can you please tell me how you do the "i'm just grabbing the theme from chakra and putting in nativebase for now" part?

Frontend definitely isn't my strength; so I'm not sure how that could be done. I would really appreciate your help on this :)

**Comment by iamclaytonray** [comment]
Sure thing.

If you grab values from Chakra's [theme](https://github.com/chakra-ui/chakra-ui/tree/main/packages/theme/src), you can put them in your custom NativeBase [theme](https://docs.nativebase.io/customizing-theme).

It's not a 100% 1:1 match so you'll need to compare the two theme objects and make changes where relevant. For example, NativeBase supports different color objects (compare the keys in both libraries' theme) so make sure you put your color palette under the correct key in NativeBase coming from Chakra. 


For components, it's mostly the same process. Just extend the NativeBase theme and use the values in Chakra's components theme (like border radius, etc)

**Comment by pierroo** [comment]
Thank you so much @iamclaytonray , very appreciate you taking the time to share more details about it.
I'm a bit confused on how to do it exactly right now, but I will definitely do my part and investigate further on it.

And I would like - just like you did - to emphasize that this is only a temporary hack until we can use our beloved chakra ui on react native. :)

**Comment by merganon** [comment]
Maybe you guys can team up with NativeBase? They already did the initial work of providing modals etc., but they definitely need some help with design. At the current moment it doesn't look very polished - like a cheap copy of Chakra UI.

I tried so many React Native component libraries, nothing comes close to the beauty of Chakra UI 😢

**Comment by iamclaytonray** [comment]
@segunadebayo - I see the help wanted label on here. Is there some process to become a collab for RN?

**Comment by redbar0n** [comment]
I just found TamagUI[TamagUI](https://tamagui.dev/) (prev called SnackUI) which is a recent alternative - to ChackraUI, Dripsy, MagnusUI, and NativeBase - that supports React Native (+Web) and is able to do responsivity + SSR. It is fairly complex, using Webpack and Babel, to compile down to web primitives and atomic CSS. I hope knowledge about Tamagui may help other users here, and help the maintainers for inspiration and insight into the complexity of and approaches to bridging the two environments (native and web).

**Comment by HommeSauvage** [comment]
TamagUI doesn't work at all (try to get any expo app to work with this thing, endless errors and very badly typed). Dripsy isn't as elegant as ChakraUI. Shopify's restyled is too rigid. We really need ChakraUI for RN.

**Comment by natew** [comment]
@HommeSauvage someone pointed me here but we did regress types and the native examples at one point, but both are fixed now.

**Comment by merganon** [comment]
Anyone using NativeBase in production?

**Comment by JClackett** [comment]
Other than the not so pretty default styles, NativeBase is the closest thing there is to a chakra in react native by far. Been using it in our teams [boilerplate](https://github.com/NoQuarterTeam/boilerplate) for a while

**Comment by AhsanSarwar45** [comment]
NativeBase is nice but we have been experiencing serious performance issues on Android and iOS with it, which makes it unsuitable for use in production on mobile devices currently. Really missing Chakra in RN.

**Comment by merganon** [comment]
> NativeBase is nice but we have been experiencing serious performance issues on Android and iOS with it, which makes it unsuitable for use in production on mobile devices currently. Really missing Chakra in RN.

Yeah seems like NativeBase has performance issues especially with Android.

https://github.com/GeekyAnts/NativeBase/issues/4302
https://github.com/GeekyAnts/NativeBase/issues/4681

Honestly, I'm thinking about just using Tailwind. There is no good UI component for React Native out there. If Chakra decides to launch a version for React Native it would literally change the world

**Comment by jukialen** [comment]
What will Chakra UI be on React Native too? 🤔

**Comment by mrjackdavis** [comment]
I've been wandering around the web for days looking for a library that'll give me `chakra-ui`'s styled system for React Native. Everywhere I turn, there are compromises I don't want to make.

WDYT about trying to deliver `@chakra-ui/react-native` as a first step? Specifically, if we had the [Chakra factory function](https://chakra-ui.com/docs/styled-system/features/chakra-factory#chakra-factory-function) for react-native, it'd be in a pretty useable state 

I'd be happy to have a go at that if it sounds like a good step forward

**Comment by itsmelion** [comment]
+1
Exactly, I am really mostly interested in the factory.

**Comment by rafarc21** [comment]
+1 on this @mrjackdavis !

**Comment by natew** [comment]
Check out the Tamagui beta, very closely aligned in terms of features with a lot of nice upgrades. Animations, even with reanimated and AnimatePresence, nested themes, nested styled factory functions that type properly. Plus a size prop that works across most components to scale, and the optimizing compiler. I landed at least 20 type improvements and it feels great. 

```
npx create-tamagui-app@latest
```

Sets you up with solito, next, expo, fonts, themes, animations all ready to go.

**Comment by iamclaytonray** [comment]
FWIW, Expo is complete garbage in my experience and opinion. As a RN dev of over 4 years, it still falls flat on its face.

Tamagui seems decent but tying that into Expo was the wrong move for any complex app. Simple todo apps I suppose is fine but anything complex, it's a joke

**Comment by nandorojo** [comment]
with the latest versions of expo, you are basically just using bare RN with plugins that make installing native code easier. you can use any native libs now, and it doesn’t come with all the bloated expo libraries anymore — only the ones you use. the only different with the RN cli is well-maintained packages and an easier DX for deploying a bare app

**Comment by wobsoriano** [comment]
+1 to Tamagui
+1 to Dripsy

**Comment by santigp258** [comment]
Have you tried [nativsebase](https://nativebase.io/)

**Comment by iamclaytonray** [comment]
Expo docs makes it very clear they still hijack the runtime, have limited native capabilities (including using the frameworks permission layers instead of native permission layers), etc. I don't see how Expo has changed at all. Hopefully they removed all of the data collection that both Apple and Google continuously flagged apps for in the past. 

Either way, seems like decent UI. If it ever gets decoupled from Expo, I'll give it a shot

**Comment by natew** [comment]
It’s not coupled to Expo at all. It’s just much easier to start from there. I use it unplugged personally on my own project.

**Comment by phuongwd** [comment]
> +1 to Tamagui +1 to Dripsy

+1 to https://github.com/Shopify/restyle

**Comment by merganon** [comment]
> Have you tried [nativebase](https://nativebase.io/)?

NativeBase has performance issues as discussed here: https://github.com/chakra-ui/chakra-ui/issues/183#issuecomment-1079927856

**Comment by bulby97** [comment]
Tamagui looks very promising, but quite complex, I might try it.
NativeBase is the best fit, but have performance issues. It seems like they're actively working on it atm.
Dripsy is cool but it's just the emotion of chakra-ui: a base to work on.
Restyle is way too strict.

**Comment by natew** [comment]
@smontlouis the compiler is totally optional, you can start just by using it as a plain library.

**Comment by smontlouis** [comment]
hi @natew 
It's not about the compiler but the lib itself.

I'm used to chakra-UI, and I know the concept of tokens, variants, themes, etc.
But the way there are used and thought are what make it complicated to grasp imo

**Comment by smontlouis** [comment]
Update: After a day of reading tamagui and understanding the paradigm shift, I believe it to be the best alternative. To be fair I believe it to be a game changer, especially with the power of the compiler. 

It just needs more components, but the fact that this is already working with mobile and web + fixes css-in-js runtime issues makes it even more powerful than chakra-ui foundation (emotion). I'm still struggling to understand everything but I feel confident choosing it for my next projet. 

Congrats @natew

**Comment by wobsoriano** [comment]
New kid in town https://react-native-zephyr.vercel.app//default-theme

**Comment by dalley8626** [comment]
@iamclaytonray it's seems like bumbag is not being managed anymore

**Comment by Valar103769** [comment]
[ui-kitten](https://akveo.github.io/react-native-ui-kitten/docs/components/drawer/api#draweritem) is awesome and beautiful , but this library is postponed the maintenance , iteslf can run on `react-native: ~0.61.5` as well , 
has some small problem on  `react-native: 0.69.4`, because some api is deprecated, and i try resolve it by yarn resolutions is fail
```json
{
"resolutions":{
"@ui-kitten/components/react-native":" ~0.61.5"
}
}
```
maybe my usage way has peoblem, you can try it angin
finally, i resolve it by [patch-package](https://github.com/ds300/patch-package)

**Comment by biapar** [comment]
Wix UI lib.

**Comment by tothemarstoken** [comment]
> Anyone using NativeBase in production?

NativeBase is so slow.

**Comment by mikehuebner** [comment]
Personally enjoy the dev experience of NativeBase, very similar to Chakra with the style props. Something I was hoping would exist with tamagui.

**Comment by pierroo** [comment]
@mikehuebner careful what you wish for; we were all fooled in the first place; but spare yourself the hassle of deploying it to production. Not until they FINALLY fix their performance issue after more than a year without actual improvement.

**Comment by mikehuebner** [comment]
@pierroo You made me dig pretty far down the tamagui rabbit hole, have it running on a new project and going to create some stuff compared against NativeBase. Will get back if I'm a believer, however this is probably not the place for this. Probably in the NativeBase repo.

Would still love to see Chakra build a native part to this library and would support if that is a goal of theirs 🙂

**Comment by TowhidKashem** [comment]
+100000 dreaming for the day there's an official chakra for react native that's 1:1 with the web version.

I tried every component lib under the sub and chakra is by far the best looking and complete. Conversely there isn't a single react native component lib that looks like it isn't from 1995 😩

**Comment by natew** [comment]
@TowhidKashem not to bogart the thread but genuinely curious what you think could be better design-wise on Tamagui?

**Comment by dalley8626** [comment]
@natew although i think that Tamagui looks promising, i believe that it's still require at least another 6 to 12 months, before it's production ready. Therefore, @TowhidKashem might as well remove all the confusion and transition to tailwind css for react native. You'll see more value rather than jumping from one design system to another, i know it's a scary journey, but it'll be all worth it.

**Comment by nandorojo** [comment]
I’ve been using Tamagui recently. I’m very impressed overall.

**Comment by natew** [comment]
Multiple 1B+ market cap companies have apps in production app in both app stores already @dalley8626

**Comment by sinafath** [comment]
Tamagui doesn't support rtl

**Comment by natew** [comment]
@sinafath there was a bug on some *Start and *End style values that has since been fixed. Rtl should work.

**Comment by Developeranees** [comment]
@perjonsson Good luck and give us a detailed review about Tamagui after release

**Comment by Enfield-Li** [comment]
tamagui's starter project with expo does not seems to work due to a critical bug,

**Comment by Enfield-Li** [comment]
> New kid in town
> 
> https://github.com/FormidableLabs/react-native-zephyr

This is pretty sweet, love it

**Comment by alitnk** [comment]
> Tamagui doesn't support rtl

Hey Sina, I've used Tamagui on a Persian project (RTL) and it worked great! What was the issue you had with RTL? @sinafath

**Comment by alitnk** [comment]
> tamagui's starter project with expo does not seems to work due to a critical bug,

Hey @Enfield-Li, I'm trying to reproduce the issue with starter but I'm unable to. Would you mind sharing the error you got and how you got it?

**Comment by ntorionbearstudio** [comment]
Hello folks, with my team BearStudio we are working on a new UI library for React Native, Ficus UI, that is inspired by Chakra UI and based on Magnus UI (other react native lib) : 

https://ficus-ui.com/
https://github.com/BearStudio/react-native-ficus-ui

The idea is to provide the Chakra UI simple components on react native. Take a look if you want and give us feedbacks, it's in alpha for the moment
